src/requests/cache/sorted_set/sorted_set_put_elements.rs

```diff
@@ -1,17 +1,104 @@
+use std::collections::HashMap;
+
 use momento_protos::cache_client::{SortedSetElement, SortedSetPutRequest};
 
 use crate::requests::cache::MomentoRequest;
 use crate::simple_cache_client::prep_request_with_timeout;
 use crate::{CacheClient, CollectionTtl, IntoBytes, MomentoResult};
 
+/// This trait defines an interface for converting a type into a vector of `SortedSetElement`s.
+pub trait IntoSortedSetElements: Send {
+    /// Converts the type into a vector of `SortedSetElement`s.
+    fn into_sorted_set_elements(self) -> Vec<SortedSetElement>;
+}
+
+/// Converts an iterator of value-score pairs into a vector of `SortedSetElement`s.
+///
+/// # Arguments
+///
+/// - `iter`: An iterator over pairs `(V, f64)` where `V` is a value that implements
+///   the `IntoBytes` trait, and `f64` represents the score associated with the value.
+///
+/// # Returns
+///
+/// A `Vec<SortedSetElement>` where each `SortedSetElement` contains a byte representation
+/// of the value and its associated score. The order of elements in the returned vector
+/// matches the order of pairs in the input iterator.
+///
+/// # Examples
+///
+/// Basic usage with a vector of tuples:
+///
+/// ```
+/// use momento::requests::cache::sorted_set::sorted_set_put_elements::map_and_collect_sorted_set_elements;
+/// let pairs = vec![("value1", 1.0), ("value2", 2.0)];
+/// let sorted_set_elements = map_and_collect_sorted_set_elements(pairs.into_iter());
+/// // `sorted_set_elements` is now a `Vec<SortedSetElement>` with byte representations of "value1" and "value2"
+/// # assert_eq!(sorted_set_elements, vec![
+/// #    momento_protos::cache_client::SortedSetElement {
+/// #        value: "value1".as_bytes().to_vec(),
+/// #        score: 1.0,
+/// #    },
+/// #    momento_protos::cache_client::SortedSetElement {
+/// #        value: "value2".as_bytes().to_vec(),
+/// #        score: 2.0,
+/// #    },
+/// # ]);
+/// ```
+///
+/// Usage with a `HashMap`:
+///
+/// ```
+/// use momento::requests::cache::sorted_set::sorted_set_put_elements::map_and_collect_sorted_set_elements;
+/// use std::collections::HashMap;
+/// let mut map = HashMap::new();
+/// map.insert("value1", 1.0);
+/// map.insert("value2", 2.0);
+/// let sorted_set_elements = map_and_collect_sorted_set_elements(map.into_iter());
+/// // `sorted_set_elements` is similar as above, suitable for sorted set operations
+/// # assert_eq!(sorted_set_elements, vec![
+/// #    momento_protos::cache_client::SortedSetElement {
+/// #        value: "value1".as_bytes().to_vec(),
+/// #        score: 1.0,
+/// #    },
+/// #    momento_protos::cache_client::SortedSetElement {
+/// #        value: "value2".as_bytes().to_vec(),
+/// #        score: 2.0,
+/// #    },
+/// # ]);
+/// ```
+pub fn map_and_collect_sorted_set_elements<I, V>(iter: I) -> Vec<SortedSetElement>
```

Comment:
This will be helpful to anyone who implements for their own collection type, hence I made public.

I wanted this to be another default function on the trait, but I could not get the `IntoIterator<(IntoBytes, f64)>` type to work as I wanted.

Reply:
let's keep this private for now. if you can figure out a way to do an Into then I'm okay with it being public (and I will look at the code a bit in a second) but I don't want to add utility functions to our public API surface area out of the gate.

Reply:
changed in [f5cd3b2](https://github.com/momentohq/client-sdk-rust/pull/207/commits/f5cd3b25d6fd398a6dfcb18be37b8228d17d1e4c)